src/Container.php

```diff
@@ -264,8 +262,11 @@ public static function create(): self
                 $container->getIndexXmlCoverageParser(),
                 $container->getXmlCoverageParser(),
             ),
-            IndexXmlCoverageLocator::class => static fn (self $container): IndexXmlCoverageLocator => new IndexXmlCoverageLocator(
-                $container->getConfiguration()->coveragePath,
+            IndexReportLocator::class => static fn (self $container): ReportLocator => new MemoizedLocator(
```

Comment:
@sanmai from what I see we cannot have that with the current container.

I notice we have some code where we have `ServiceFoo` and then `MemoizedServiceFoo`, but if feels quite boilertemplaty. Is there any way we could get rid of that?

Looks like it's a sanity check that is done, and tbh I'd argue it makes sense most of the time, but maybe it could either be disabled for a specific service or extended to allow this case for interfaces (i.e. whilst the output is not the same instance as the ID, it is still of the same parent type).

Reply:
Well in this case you cannot have `MemoizedLocator` as an ID as there would be two services with that ID...